### PR TITLE
Package template to compile according to Linux type

### PR DESCRIPTION
**Pull request by MarcelKemp**
|Related issue|
|---|
|#8167|

## Description

In this PR, a linux package template has been created to decide which functions and libraries are compiled according to the type of Linux it is.

Currently it has been divided between functions for all Linux (`rpm` and `dpkg`), and extra functions (`pacman`).

When compiling, the chosen Linux type will be decided based on the `LINUX_TYPE` variable. And in the case of _CentOS 5_, as it is not possible to compile the `pacman` libraries because of the kernel version (https://github.com/wazuh/wazuh/issues/8167#issuecomment-823469535), then it has been decided to compile the `rpm` and `dpkg` functions, while for the rest of Linux the three functions are compiled.

## Logs/Alerts example

```
sysinfo_test_tool:

ArchLinux - pacman
[{"architecture":"x86_64","description":"Access control list utilities, libraries and headers","format":"pacman","groups":"","install_time":"2021/03/01 13:06:56","name":"acl","size":331256,"vendor":"Arch Linux","version":"2.2.53-3"},{"architecture":"any","description":"Monospaced font family for user interface and coding environments","format":"pacman","groups":"","install_time":"2021/03/10 17:19:55","name":"adobe-source-code-pro-fonts","size":1945214,"vendor":"Arch Linux","version":"2.038ro+1.058it+1.018var-1"},{"architecture":"any","description":"GNOME standard icons","format":"pacman","groups":"","install_time":"2021/03/10 17:19:55","name":"adwaita-icon-theme","size":23457529,"vendor":"Arch Linux","version":"3.38.0-1"}...

Debian 7 - deb
[{"architecture":"all","description":"internationalized substitute of Text::Wrap","format":"deb","groups":"perl","multiarch":"","name":"libtext-wrapi18n-perl","priority":"required","size":28,"source":"","vendor":"Anibal Monsalve Salazar <anibal@debian.org>","version":"0.06-7"},{"architecture":"amd64","description":"Micro string library: shared library","format":"deb","groups":"libs","multiarch":"same","name":"libustr-1.0-1","priority":"optional","size":279,"source":"ustr","vendor":"Vaclav Ovsik <vaclav.ovsik@i.cz>","version":"1.0.4-3"}...

CentOS 5 - rpm
[{"architecture":"noarch","description":"A set of system configuration and setup files.","format":"rpm","groups":"System Environment/Base","install_time":"1425472725","name":"setup","size":381527,"vendor":"CentOS","version":"2.5.58-9.el5"},{"architecture":"noarch","description":"The skeleton package which defines a simple CentOS system.","format":"rpm","groups":"System Environment/Base","install_time":"1425472726","name":"basesystem","size":0,"vendor":"CentOS","version":"8.0-5.1.1.el5.centos"}...


```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade

### CODE REVIEW

**Review by Dwordcito — APPROVED** [review]
Looks good! Congratulations on the code.

File: src/data_provider/src/packages/packageLinuxDataRetriever.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @param jsonPackages JSON to be filled
+ */
+void    getRpmInfo(nlohmann::json& jsonPackages);
+
+/**
+ * @brief Fills a JSON object with all available dpkg-related information
+ * @param libPath Path to dpkg's database directory
+ * @param jsonPackages JSON to be filled
+ */
+void    getDpkgInfo(const std::string& libPath, nlohmann::json& jsonPackages);
```
Comment (konangc87) [inline src/data_provider/src/packages/packageLinuxDataRetriever.h:37]: pls remove extra spaces

**Review by konangc87 — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
LGTM!

**Review by chemamartinez — APPROVED** [review]

**Review by konangc87 — APPROVED** [review]
LGTM!

**Pull request merged**